src/main/cpp/util/file_windows.cc

```diff
@@ -887,10 +724,15 @@ bool CanAccessDirectory(const std::string& path) {
 }
 
 bool IsDirectoryW(const wstring& path) {
-  DWORD attrs = ::GetFileAttributesW(path.c_str());
-  return (attrs != INVALID_FILE_ATTRIBUTES) &&
-         (attrs & FILE_ATTRIBUTE_DIRECTORY) &&
-         JunctionResolver().Resolve(path.c_str(), nullptr);
+  // Attempt opening the path, which may be anything -- a file, a directory, a
+  // symlink, even a dangling symlink is fine.
+  // Follow reparse points in order to return false for dangling ones.
+  AutoHandle h(CreateFileW(path.c_str(), 0, kAllShare, NULL, OPEN_EXISTING,
+                           FILE_FLAG_BACKUP_SEMANTICS, NULL));
+  BY_HANDLE_FILE_INFORMATION info;
+  return h.IsValid() && GetFileInformationByHandle(h, &info) &&
```

Comment:
Nit: This performance of this function is much slower than reading the reparse point directly. But I'm fine using this for resolving for junction, but not for symlink. Because we use the symlink resolver in https://github.com/bazelbuild/bazel/blob/master/src/main/tools/build-runfiles-windows.cc

Reply:
Thanks for the information!

With this PR we will use it for `RealPath`, `PathExists`, and `IsDirectory`.